utils/ansible/ftest/library/daos_hugepages.py

```diff
@@ -0,0 +1,143 @@
+
+version_added: "0.1"
+
+description:
+    - Check if the Linux Huge Pages feature is enabled
+    - Check the value of the Huge Pages
+    - Set the value of the Huge Pages if needed
+
+options:
+    size:
+        description:
+            - Number of huge pages to allocate
+        type: int
+        default: 4096
+
+author:
+    - DAOS group <daos@daos.groups.io>
+'''
+
+EXAMPLES = '''
+- name: Set the Linux Huge Pages size
+  daos_hugepages
+'''
+
+RETURN = '''
+elapsed:
+     description: the number of seconds that elapsed while checking and setting the huge pages size
+     returned: always
+     type: int
+     sample: 3
+'''
+
+import os
+import re
+import datetime
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+
+__HUGEPAGES_STATE_PATTERN__ = re.compile(r"(\[madvise\])|(\[always\])")
+__HUGEPAGES_SYSCTL_PATTERN = re.compile(r"^vm\.nr_hugepages\s*=\s*(?P<size>\d+)$")
+
+def is_huge_pages_enabled():
+    with open(r'/sys/kernel/mm/transparent_hugepage/enabled', 'r') as fd:
+        line = fd.read()
+        return __HUGEPAGES_STATE_PATTERN__.match(line)
+
+def main():
+    """Ansible module setting the Huge Pages size."""
+
+    ### Default Ansible module prologue ###
+    argument_spec = dict(size=dict(type='int', default=4096))
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+    check_mode = module.check_mode
+    hugepages_size = module.params['size']
+
+    ### Huge Pages configuration
+    start_time = datetime.datetime.utcnow()
+    if not os.path.isfile(r"/sys/kernel/mm/transparent_hugepage/enabled"):
+        module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                msg="Huge Pages not activated in kernel")
+
+    if not is_huge_pages_enabled():
+        try:
+            with open(r"/sys/kernel/mm/transparent_hugepage/enabled", 'w') as fd:
+                fd.write('madvise')
+        except Exception as ex:
+            module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                    msg='Huge Pages could not be enabled: {}'.format(ex))
+        if not is_huge_pages_enabled():
+            module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                    msg="Huge Pages could not be enabled")
+
+    cp = subprocess.run([r'sysctl', r'vm.nr_hugepages'],
+            stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=3)
+    if cp.returncode != 0:
+        module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                msg="Size of Huge Pages could not be read: {}".format(cp.stderr.decode('ascii')))
+
+    hugepages_current_size = 0
+    stdout_str = cp.stdout.decode('ascii')
+    m = __HUGEPAGES_SYSCTL_PATTERN.match(stdout_str)
+    if m is None:
+        module.fail_json(elapsed=(datetime.datetime.utcnow() - start_time).seconds,
+                msg="Invalid size of huge pages from sysctl: {}".format(stdout_str))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string